lib/ui/painting/picture.cc

```diff
@@ -16,6 +18,64 @@
 #include "third_party/tonic/logging/dart_invoke.h"
 
 namespace flutter {
+namespace {
+
+sk_sp<SkSurface> MakeSnapshotSurface(const SkISize& picture_size,
+                                     fml::WeakPtr<GrContext> resource_context) {
+  SkImageInfo image_info = SkImageInfo::MakeN32Premul(
+      picture_size.width(), picture_size.height(), SkColorSpace::MakeSRGB());
+  if (resource_context) {
+    return SkSurface::MakeRenderTarget(resource_context.get(),  // context
+                                       SkBudgeted::kNo,         // budgeted
+                                       image_info               // image info
+    );
+  } else {
+    return SkSurface::MakeRaster(image_info);
+  }
+}
+
+/// Makes a RAM backed (Raster) image of a picture.
+/// @param[in] picture The picture that will get converted to an image.
+/// @param[in] surface The surface tha will be used to render the picture.  This
+///                    will be CPU or GPU based.
+/// @todo Currently this creates a RAM backed image regardless of what type of
```

Comment:
"RAM backed" is odd terminology. Even device memory is in RAM right? Can we stick to just the device/host terminology like you have done for the traces?

Reply:
IMO device/host is very confusing.  Even yuqian asked me to add a comment to his code to clarify them.  Images in RAM vs VRAM is the model I've used when dealing with the discrete GPU model.  Can you tell me exactly what you want here and I'll plop it in.